scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala

```diff
@@ -229,7 +229,13 @@ abstract class FileSource extends SchemedSource with LocalSourceOverride with Hf
    * TODO: consider writing a more in-depth version of this method in [[TimePathedSource]] that looks for
    * TODO: missing days / hours etc.
    */
-  protected def pathIsGood(globPattern: String, conf: Configuration) = FileSource.globHasNonHiddenPaths(globPattern, conf)
+  protected def pathIsGood(globPattern: String, conf: Configuration) = {
+    if (conf.getBoolean("scalding.require_success_file", false)) {
```

Comment:
uh this seems fine given this is a cherrypick pr... but probably should be using a constant?

Reply:
we didn't want to break binary compatibility, but a constant would be a nice follow up.